client_plugin/drivers/shared/shared_driver.go

```diff
@@ -62,6 +86,18 @@ func NewVolumeDriver(cfg config.Config, mountDir string) *VolumeDriver {
 	}
 	d.dockerd = cli
 
+	// Pull Samba image
+	log.Infof("Pulling Samba server image")
```

Comment:
Was thinking about that. But downloading during plugin creation would mean that the samba server will become a part of the plugin.

We want to run it outside of plugin as a docker service. Correct me if I am wrong.